telegram/utils/helpers.py

```diff
@@ -40,54 +44,148 @@ def get_signal_name(signum):
+    # making every isinstace() call more delicate. So we just use our own compat layer.
+    def _datetime_to_float_timestamp(dt_obj):
+        epoch_dt = EPOCH_DT if dt_obj.tzinfo is not None else NAIVE_EPOCH_DT
+        return (dt_obj - epoch_dt).total_seconds()
 
+_datetime_to_float_timestamp.__doc__ = \
+    """Converts a datetime object to a float timestamp (with sub-second precision).
+If the datetime object is timezone-naive, it is assumed to be in UTC."""
 
-def to_timestamp(dt_obj):
+
+def to_float_timestamp(t, reference_timestamp=None):
     """
+    Converts a given time object to a float POSIX timestamp.
+    Used to convert different time specifications to a common format. The time object
+    can be relative (i.e. indicate a time increment, or a time of day) or absolute.
+    Any objects from the :module:`datetime` module that are timezone-naive will be assumed
+    to be in UTC.
+
+    ``None`` s are left alone (i.e. ``to_float_timestamp(None)`` is ``None``).
+
     Args:
-        dt_obj (:class:`datetime.datetime`):
+        t (int | float | datetime.timedelta | datetime.datetime | datetime.time | None):
+            Time value to convert. The semantics of this parameter will depend on its type:
+
+            * :obj:`int` or :obj:`float` will be interpreted as "seconds from ``reference_t``"
+            * :obj:`datetime.timedelta` will be interpreted as
+              "time increment from ``reference_t``"
+            * :obj:`datetime.datetime` will be interpreted as an absolute date/time value
+            * :obj:`datetime.time` will be interpreted as a specific time of day
+
+        reference_timestamp (float, optional): POSIX timestamp that indicates the absolute time
+            from which relative calculations are to be performed (e.g. when ``t`` is given as an
+            :obj:`int`, indicating "seconds from ``reference_t``"). Defaults to now (the time at
+            which this function is called).
 
     Returns:
-        int:
+        (float | None) The return value depends on the type of argument ``t``. If ``t`` is
+            given as a time increment (i.e. as a obj:`int`, :obj:`float` or
+            :obj:`datetime.timedelta`), then the return value will be ``reference_t`` + ``t``.
 
+            Else if it is given as an absolute date/time value (i.e. a :obj:`datetime.datetime`
+            object), the equivalent value as a POSIX timestamp will be returned.
+
+            Finally, if it is a time of the day without date (i.e. a :obj:`datetime.time`
+            object), the return value is the nearest future occurrence of that time of day.
+
+            If ``t`` is ``None``, ``None`` is returned (to facilitate formulating HTTP requests
+            when the object to be serialized has a ``date`` which is ``None`` without having to
+            check explicitly).
     """
-    if not dt_obj:
+
+    if reference_timestamp is None:
+        reference_timestamp = time.time()
+
+    if t is None:
```

Comment:
I was also keen to remove that, but then I saw that a good chunk of calls to `to_timestamp`, i.e. when making a dict from a `telegram` API object (to be then transformed into JSON), rely on this behaviour (so that it's less verbose: `to_timestamp(x)` rather than `to_timestamp(x) if x is not None else None`. For example, see https://github.com/python-telegram-bot/python-telegram-bot/blob/965ad17af810df0a4c04ce37f335e78d406a2f69/telegram/chatmember.py#L156-L159

So, to keep it consistent, I kept the behaviour with `to_float_timestamp`  too. 

Should I make a separate function that wraps `to_timestamp` with this `None` check, e.g. `maybe_to_timestamp`, and remove it from `to_timestamp` and `to_float_timestamp`?

Reply:
that sounds like a good solution. lets to do it with a wrapper as you've suggested

Reply:
Ok. Technically this is a breaking change, since the name of `to_timestamp` suggests it's public API; here's hoping that nobody relied on that behaviour. Should I name the wrapper with a leading underscore? And is `_maybe_to_timestamp` a good name? I can't think of another one.

Reply:
You've got a good point regarding the naming and bkwd-compat. Thanks.
 Looking at the code as it is now, admitting past mistakes but yet considering backward compatibility, I think we should stick with:

1. New function: `to_float_timestamp()` - will raise `TypeError` if `t` is `None`. (the same `raise` you've put at the end of the function.
2. Older functions: `to_timestamp()` & `from_timestamp()` - since both had previously returned `None` for `None` input, their names will remain the same and they will continue accepting `None`. (so we won't be using `maybe_XXX_timestamp()` afterall, even though I really liked it).

Reply:
Change made in 82fe342.